dawn/src/dawn/Optimizer/TemporaryHandling.cpp

```diff
@@ -146,10 +146,12 @@ void demoteTemporaryFieldToLocalVariable(iir::StencilInstantiation* instantiatio
   DAWN_ASSERT_MSG(assignmentExpr,
                   "first access of field (i.e lifetime.Begin) is not an `AssignmentExpr`");
 
-  // Create the new `VarDeclStmt` which will replace the old `ExprStmt`
-  std::shared_ptr<iir::Stmt> varDeclStmt =
-      iir::makeVarDeclStmt(Type(BuiltinTypeID::Float), varname, 0, "=",
-                           std::vector<std::shared_ptr<iir::Expr>>{assignmentExpr->getRight()});
+  // Remove the field
+  instantiation->getMetaData().removeAccessID(AccessID);
+
+  // Create the new `VarDeclStmt` which will replace the old `ExprStmt` and register the variable
+  std::shared_ptr<iir::Stmt> varDeclStmt = instantiation->getMetaData().declareVar(
+      true, varname, Type(BuiltinTypeID::Float), assignmentExpr->getRight(), AccessID);
```

Comment:
so I'm not sure I understand the keep var name thing. Why is it true here, but false in other instances you `declareVar`? is this related to the (broken) `keepVarNames` flag?